orchestrator/api/helpers.py

```diff
@@ -27,11 +28,30 @@
 
 from orchestrator.api.error_handling import raise_status
 from orchestrator.db import ProcessTable, ProductTable, SubscriptionTable
+from orchestrator.db.models import SubscriptionSearchView
 from orchestrator.domain.base import SubscriptionModel
 
 logger = get_logger(__name__)
 
 
+def _quote_if_kv_pair(token: str) -> str:
+    return f'"{token}"' if ":" in token else token
+
+
+def _process_text_query(q: str) -> str:
+    quote = '"'
+    if q.count(quote) % 2 == 1:
+        q += quote  # Add missing closing quote
+    lex = shlex(q)
+    lex.whitespace_split = True
+    lex.quotes = '"'
+    try:
+        return " ".join([_quote_if_kv_pair(token) for token in lex])
+    except ValueError:
```

Comment:
Can this really fail?

Reply:
Could test this with `hypothesis` with a string strategy

Reply:
But I don't see the use of logging an error here, because users can type anything into the url.
I'd say log it as debug